sct-commons/src/test/java/org/lfenergy/compas/sct/commons/testhelpers/SclObjectFactoryHelper.java

```diff
@@ -0,0 +1,45 @@
+// SPDX-FileCopyrightText: 2022 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons.testhelpers;
+
+import lombok.experimental.UtilityClass;
+import org.lfenergy.compas.scl2007b4.model.*;
+
+import java.math.BigDecimal;
+
+/**
+ * Provides static methods to quickly retrieve SCL elements, to be used in writing tests.
+ * Methods throw AssertionFailedError when element not found (equivalent of calling JUnit Assert.fail())
+ * which stops the test nicely (compared to an exception error), with a message and a failed status.
+ */
+@UtilityClass
+public class SclObjectFactoryHelper {
+
+    public static TConnectedAP addConnectedAp(SCL scd, String apName, String iedName) {
+        scd.setCommunication(new TCommunication());
+        TSubNetwork subNetwork = new TSubNetwork();
+        scd.getCommunication().getSubNetwork().add(subNetwork);
+        TConnectedAP connectedAP = new TConnectedAP();
+        subNetwork.getConnectedAP().add(connectedAP);
+        connectedAP.setApName(apName);
+        connectedAP.setIedName(iedName);
+        return connectedAP;
+    }
+
+    public static TDurationInMilliSec newDurationInMilliSec(String value) {
+        TDurationInMilliSec minTime = new TDurationInMilliSec();
+        minTime.setUnit("s");
+        minTime.setMultiplier("m");
+        minTime.setValue(new BigDecimal(value));
+        return minTime;
+    }
+
+    public static TP newP(String type, String value){
```

Comment:
Could be moved in utils or a common class cause is duplicate with the one in ControlBlockAdapter